File: check_profiles.js

import { createClient } from '@supabase/supabase-js';

const supabaseUrl = "https://lqdxsgnoeickomhsgeco.supabase.co";
const anonKey = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6ImxxZHhzZ25vZWlja29taHNnZWNvIiwicm9sZSI6ImFub24iLCJpYXQiOjE3Njc4NTIwMDEsImV4cCI6MjA4MzQyODAwMX0.UtOsTUJS2Z8uqi7pzDqn8eZdv9T5CvZRVGDsOrC_gms";

const supabase = createClient(supabaseUrl, anonKey);

async function checkProfiles() {
    const { data, error } = await supabase.from('profiles').select('*');
    if (error) console.log("Profiles Error:", error);
    else console.log(`Profiles Found: ${data.length}`);
}

checkProfiles();

File: backend/db/check_profiles.py
import os
from dotenv import load_dotenv
from supabase import create_client

load_dotenv()
url = os.getenv("VITE_SUPABASE_URL")
key = os.getenv("SUPABASE_SERVICE_ROLE_KEY")
supabase = create_client(url, key)

print("--- User Profiles ---")
try:
    up = supabase.table("user_profiles").select("*").execute()
    print(f"Total User Profiles: {len(up.data)}")
    for p in up.data:
        print(f"- {p.get('email')} ({p.get('id')})")
except Exception as e:
    print(f"Error user_profiles: {e}")

print("\n--- Auth Users (via Service Role) ---")
# Note: This might not work via standard client if it doesn't have auth admin access
try:
    # We can try to count auth table if possible or just assume from profiles
    pass 
except:
    pass
